homeassistant/components/kostal_plenticore/select.py

```diff
@@ -12,22 +12,29 @@
 from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
+from ...helpers.entity_platform import AddEntitiesCallback
 from .const import DOMAIN, SELECT_SETTINGS_DATA
 from .helper import Plenticore, SelectDataUpdateCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_entry(
-    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Add kostal plenticore Select widget."""
     plenticore: Plenticore = hass.data[DOMAIN][entry.entry_id]
+    select_data_update_coordinator = SelectDataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        "Settings Data",
+        timedelta(seconds=30),
+        plenticore,
+    )
 
     async_add_entities(
         PlenticoreDataSelect(
-            hass=hass,
-            plenticore=plenticore,
+            select_data_update_coordinator,
```

Comment:
Do we want to use the same coordinator for all these entities. Before we created a new coordinator per entity.

Reply:
Yes, before change was driven by my unexperience and a previous change request. Now its back conistent with switch.py and sensor.py